fs_storage/wizards/fs_test_connection.py

```diff
@@ -0,0 +1,28 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class FSTestConnection(models.TransientModel):
+    _name = "fs.test.connection"
+    _description = "FS Test Connection Wizard"
+
+    def _get_check_connection_method_selection(self):
+        return self.env["fs.storage"]._get_check_connection_method_selection()
+
+    storage_id = fields.Many2one("fs.storage")
+    check_connection_method = fields.Selection(
+        selection="_get_check_connection_method_selection",
+        required=True,
+    )
+
+    @api.model
+    def default_get(self, field_list):
+        res = super().default_get(field_list)
+        res["storage_id"] = self.env.context.get("active_id", False)
+        return res
+
+    def action_test_config(self):
+        return self.storage_id.with_context(
+            force_connection_method=self.check_connection_method
+        )._test_config()
```

Comment:
Why do you pass the forced method into the context? It could be an optional argument of the `test_config` method...

```suggestion
        return self.storage_id._test_config(force_check_method=self.check_connection_method)
```

Reply:
It is because I did not want to pass an arg to `fs` property method. And connection is checked by calling the fs property...
And I did not especially want to change the way connection is checked.

Reply:
but you could also call
``` python
self.storage_id._check_connection_with_method(self.storage_id.fs, self.check_connection_method)
```
The same effect and no need to use the context to pass an args... I prefer explicit way to pass args...

Reply:
Well yes, I could, this will then call `_check_connection_with_method` twice actually : 
Once because of `self.storage_id.fs`, which will call the `_check_connection_with_method` and the second time because we call it directly.
I don't want to duplicate the code of `_test_config` so it behave differently in case it is called from wizard or from a fs.storage with a `check_connection_method`configured...

Then I see 2 options if we really get rid of the context : 
1. Always call `_check_connection_with_method` in case of manual check, it will check twice if the fs.storage is configured to have a method but it is no big deal since it only happens on manual action (which is rarely done)
2. When checking the connection manually, we always open the wizard, even if the `check_connection_method` is set.
I would rather not, because it adds an extra step and it give the choice to choose a different `check_connection_method` in the wizard than the one configured on the `fs.storage`...

I also prefer explicit way to pass args, but it seemed like the easiest way to avoid too much code duplication/addition.

Anyway, I am not against the 1 I did a commit toward this way.